### ISSUE TITLE
Patterns of different length in the same song

### ISSUE THREAD

**Issue by gutte** [hydrogen-music/hydrogen#691]
Hydrogen version *          : 0.9.7
Operating system + version  : Debian 9.6
Audio driver + version      : 

- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
Hi,

I have a scenario where I have a 1 bar (size 8) groove and a 2 bar (size 16)ride pattern (the ride should play out over two repetitions of the groove).

Groove:

![image](https://user-images.githubusercontent.com/767995/51697051-96510080-200f-11e9-9ad5-974d52cfbeb1.png)

Ride:

![image](https://user-images.githubusercontent.com/767995/51697073-a7017680-200f-11e9-9ba5-2cd071cd892c.png)

Now in **song mode**:
When i play the ride only it plays the whole pattern over the 1 "bar" seen in the song editor (size 16 -> 1 song bar):

![image](https://user-images.githubusercontent.com/767995/51697203-f0ea5c80-200f-11e9-9a14-4c7882a1bced.png)

However, when I add the groove, size 8 maps to 1 "bar" of the song:

![image](https://user-images.githubusercontent.com/767995/51697429-73731c00-2010-11e9-93a0-809dcda08766.png)

This is ok i suppose, BUT the ride then starts repeating after the first bar, and there is no way to get the whole ride pattern into the song.

I would expect to be able to do this to hear the whole ride:

![image](https://user-images.githubusercontent.com/767995/51697504-a1586080-2010-11e9-81ec-7cc7bb21ceb1.png)

The only solution I can find is to double the length of the groove, which may be simple enough in my case, but would be very tedious if I had many patterns that were shorter than my longest one.

Is there something I have missed?

In my view the underlying problem is that discussed in [issue 257](https://github.com/hydrogen-music/hydrogen/issues/257): **The size of the pattern should be defined unambiguously in bars, and those bars should be a 1:1 with the bars in the song editor.**

However, this issue could also be fixed independently by allowing for a longer pattern to continue for multiple bars in the song.
- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -

*  H2 version : see 'Info - About', or in case you build Hydrogen from source use 'git rev-parse HEAD'
** if you are reporting a bug, please describe how the bug can be reproduced

**Comment by thijz** [comment]
i dont think you are missing anything : patterns of different duration dont mix very well in H2
i always use patterns of the same duration to work around this

**Comment by tc-ubnt** [comment]
I would very welcome to have bar size matching the pattern length so I could put more bars with length 8 while playing other pattern with length 16, etc... I'm afraid that having the same bar for patterns of any length is misleading.

**Comment by theGreatWhiteShark** [comment]
Hey @gutte,

Could you try the version of Hydrogen in current master and tell if the solution provided #984 does work for you?

**Comment by cme** [comment]
Maybe also worth noting that in the current master version of Hydrogen, notes in a pattern can be copied and dragged (as pattern cells in the song editor can) which should make doubling the length of even a complex pattern an easy task:

  * Change length using drop-down
  * Select all notes with Ctrl-A, or drag selection box round them
  * either:
      * Ctrl+drag notes to move and duplicate, or
      * Ctrl+V to paste, then drag into place

**Comment by elyurn** [comment]
Hello, I have the same problem while working on polymeters.

For example, I can't write what she is doing, while I would like to hear together patterns of sizes 3-4-5-7: https://www.youtube.com/watch?v=q2nm4ho62s8

In Hydrogen, if I try to write 4 different patterns with different sizes and use them in a song, it is impossible because the software lengthen of shorten some patterns!

Would you have a solution for that please? Thank you very much for all your (great) work!

**Comment by theGreatWhiteShark** [comment]
Hey @elyurn ,

Right now we only support a mix of pattern sizes that have a common denominator. So, it's either expanding the e.g. 3-4 pattern to a length of 12 or to use successive patterns of maximal size and copy paste the corresponding parts of your polyrhythmic patterns. For a 3-4-5-7 scenario only the later is feasible (while being quite dirty and cumbersome). But with the new [Hydrogen 1.1](https://github.com/hydrogen-music/hydrogen/releases/tag/1.1.0-beta1) copy&pasting does at least becomes very easy and fast.

**Comment by theGreatWhiteShark** [comment]
By now the original issue is more or less solved. If there are multiple patterns in a single column, the length of the bar will be determined by the longest one. In addition, the UI does only fill the pattern grid partially in case a pattern is shorter than the longest one. It should be rather clear what is going on and all notes will be played back.

I willl close this issue for now. If the current solution does not fit your needs, please feel free to reopen it.

> Hello, I have the same problem while working on polymeters.
>
> For example, I can't write what she is doing, while I would like to hear together patterns of sizes 3-4-5-7: https://www.youtube.com/watch?v=q2nm4ho62s8
>
> In Hydrogen, if I try to write 4 different patterns with different sizes and use them in a song, it is impossible because the software lengthen of shorten some patterns!
 
Polyrhythm is yet a different story. Hydrogen will indeed lengthen shorter pattern but shouldn't shorten any patterns.

**Comment by gutte** [comment]
Great work, my thanks to all contributors <3.
